mantidimaging/core/operations/rebin/rebin.py

```diff
@@ -44,31 +43,26 @@ def filter_func(images: ImageStack, rebin_param=0.5, mode=None, progress=None) -
 
         :return: The processed 3D numpy.ndarray
         """
-        h.check_data_stack(images)
-
+        # Validate rebin_param
         if isinstance(rebin_param, tuple):
-            param_valid = rebin_param[0] > 0 and rebin_param[1] > 0
+            new_shape = rebin_param
+        elif isinstance(rebin_param, (int, float)):
+            current_shape = images.data.shape[1:]
+            new_shape = (int(current_shape[0] * rebin_param), int(current_shape[1] * rebin_param))
         else:
-            param_valid = rebin_param > 0
-
-        if not param_valid:
-            raise ValueError('Rebin parameter must be greater than 0')
+            raise ValueError("Invalid type for rebin_param")
 
-        empty_resized_data = _create_reshaped_array(images, rebin_param)
-
-        f = ps.create_partial(skimage.transform.resize,
-                              ps.return_to_second_at_i,
-                              mode=mode,
-                              output_shape=empty_resized_data.array.shape[1:])
-        ps.execute(partial_func=f,
-                   arrays=[images.shared_array, empty_resized_data],
-                   num_operations=images.data.shape[0],
-                   msg="Applying Rebin",
-                   progress=progress)
-        images.shared_array = empty_resized_data
+        params = {'new_shape': new_shape, 'mode': mode}
+        ps.run_compute_func(cls.compute_function, images.data.shape[0], images.shared_array, params, progress)
 
         return images
 
+    @staticmethod
+    def compute_function(i: int, array: np.ndarray, params: dict):
+        new_shape = params['new_shape']
+        mode = params['mode']
+        array[i] = resize(array[i], output_shape=new_shape, mode=mode, preserve_range=True)
```

Comment:
```
    def compute_function(i: int, arrays: list[np.ndarray], params: dict):
        new_shape = params['new_shape']
        mode = params['mode']
        input_array, output_array = arrays
        output_array[i] = resize(input_array[i], output_shape=new_shape, mode=mode, preserve_range=True)
```

Reply:
AttributeError: 'numpy.ndarray' object has no attribute 'has_shared_memory'
i have changed the code to fix this